packages/teams-js/src/internal/liveShareHost.ts

```diff
@@ -0,0 +1,238 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts } from '../public/constants';
+
+/**
+ * @hidden
+ * Hide from docs
+ * ------
+ * Allowed roles during a meeting.
+ */
+export enum UserMeetingRole {
+  guest = 'Guest',
+  attendee = 'Attendee',
+  presenter = 'Presenter',
+  organizer = 'Organizer',
+}
+
+/**
+ * State of the current Live Share sessions backing fluid container.
```

Comment:
Should this also have a `@hidden` tag on it? Generally all the code that lives in the internal directory is hidden from public docs so I would expect to see `@hidden` throughout this file.

Reply:
If these are public interfaces that app developers are expected to interact with, I think they should be defined in a file in the public folder.

Reply:
I think if they aren't meant to be exposed to app developers, then in addition to adding the `@hidden` tag, you might also want to add the following comments as well (this helps with VS Code intellisense, letting devs know they shouldn't call the functions in their apps):

```
 * @internal
 * Limited to Microsoft-internal use 
 ```

Reply:
The public interfaces are exposed by `@microsoft/live-share`. There's a bit of a chicken and an egg here which is why I'm having to redefine things. Once everything is published I can go back in and remove some of these type definitions.

The `LiveShareHost` class needs to be internal as it's implementing the callbacks needed to host Live Share within an app like the Teams client.

I've updated everything to be marked as both `@hidden` and `@internal` although looking through otehr files in this folder there's a lot of inconsistency with this.